service-base/src/main/java/org/eclipse/hono/service/AbstractApplication.java

```diff
@@ -50,12 +48,12 @@
     protected final Logger log = LoggerFactory.getLogger(getClass());
 
     private ObjectFactory<S> serviceFactory;
-    private C config;
+    private ApplicationConfigProperties config;
```

Comment:
What about changing Autowiring on the setter to required = true instead? Would also solve the problem with inconsistent defaults that I mentioned above.